file-manager/tinyfilemanager.php.patch

```diff
@@ -0,0 +1,162 @@
+--- tinyfilemanager.old.php	2024-01-29 21:27:36.788013703 +0100
++++ tinyfilemanager.php	2024-01-31 10:19:12.944928468 +0100
+@@ -203,6 +203,10 @@
+     'en' => 'English'
+ );
+ 
++//counters for added/deleted files (for metrics)
++$files_added = 0;
++$files_deleted = 0;
++
+ if ($report_errors == true) {
+     @ini_set('error_reporting', E_ALL);
+     @ini_set('display_errors', 1);
+@@ -584,6 +588,7 @@
+ 
+          function event_callback ($message) {
+             global $callback;
++            fm_report_files();
+             echo json_encode($message);
+         }
+ 
+@@ -652,6 +657,8 @@
+         }
+ 
+         if ($success) {
++            global $files_added;
++            $files_added++;
+             event_callback(array("done" => $fileinfo));
+         } else {
+             unlink($temp_file);
+@@ -700,6 +707,8 @@
+                 if(fm_is_valid_ext($new)) {
+                     @fopen($path . '/' . $new, 'w') or die('Cannot open file:  ' . $new);
+                     fm_set_msg(sprintf(lng('File').' <b>%s</b> '.lng('Created'), fm_enc($new)));
++                    global $files_added;
++                    $files_added++;
+                 } else {
+                     fm_set_msg(lng('File extension is not allowed'), 'error');
+                 }
+@@ -1021,6 +1030,8 @@
+                         $fullPathTarget = $fullPath;
+                     }
+                     rename("{$fullPath}.part", $fullPathTarget);
++                    global $files_added;
++                    $files_added++;
+                 }
+ 
+             } else if (move_uploaded_file($tmp_name, $fullPath)) {
+@@ -1030,6 +1041,8 @@
+                         'status'    => 'success',
+                         'info' => "file upload successful"
+                     );
++                    global $files_added;
++                    $files_added = 1;
+                 } else {
+                     $response = array (
+                         'status' => 'error',
+@@ -1050,6 +1063,7 @@
+         );
+     }
+     // Return the response
++    fm_report_files();
+     echo json_encode($response);
+     exit();
+ }
+@@ -1141,6 +1155,8 @@
+ 
+         if ($res) {
+             fm_set_msg(sprintf(lng('Archive').' <b>%s</b> '.lng('Created'), fm_enc($zipname)));
++            global $files_added;
++            $files_added++;
+         } else {
+             fm_set_msg(lng('Archive not created'), 'error');
+         }
+@@ -1210,6 +1226,8 @@
+ 
+         if ($res) {
+             fm_set_msg(lng('Archive unpacked'));
++            global $files_added;
++            $files_added += fm_count_files($path);
```

Comment:
I didn't even realize there was such a feature. I am not sure if we want to count this as Added files. You may want to discuss this with @Popolechien. Expanding can be unrelated to an _upload_ (and thus an _addition_ in user's mind). Given the tool offers easy upload of multiple files at once, uploading an Archive would make more sense as a whole.

Reply:
We have confirmed together this is needed. E.g. if a user unzip a file with 120 files withing, this has to be counted as 120 additions.

Reply:
OK then 🤷‍♂️